### PR TITLE
Fix allocation of byte arrays.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/4357#issuecomment-402309050

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -8060,21 +8060,24 @@ BOOST_AUTO_TEST_CASE(create_memory_array_allocation_size)
...
 				bytes memory b3 = new bytes(256);
 				bytes memory b4 = new bytes(31);
+				uint memsize;
 				assembly {
 					d1 := sub(b2, b1)
 					d2 := sub(b3, b2)
 					d3 := sub(b4, b3)
+					memsize := msize()
 				}
+				ms = memsize <= (480 + 0x80);
```
Comment (ekpyron) [inline test/libsolidity/SolidityEndToEndTest.cpp]: Can we check for equality here instead?

Comment (chriseth) [reply test/libsolidity/SolidityEndToEndTest.cpp]: If we do, the test will break as soon as some initialization routine needs more memory.

Comment (ekpyron) [reply test/libsolidity/SolidityEndToEndTest.cpp]: Maybe that's a good thing?

Comment (chriseth) [reply test/libsolidity/SolidityEndToEndTest.cpp]: Rigth, it is as long as it is no big deal to update the expectations. This is where you come in ;)

**Comment by chriseth** [comment]
@ekpyron can you check again, please?

**Review by ekpyron — APPROVED** [review]
Looks fine to me now!

**Pull request merged**